### PR TITLE
Behavior improvement

### PR DESCRIPTION
**Pull request by burzum**

### CODE REVIEW

**Comment by burzum** [comment]
Refs https://github.com/FriendsOfCake/search/issues/53

**Comment by ADmad** [comment]
This change will break usages where users used `search` option directly like:

```php
find('search', ['search' => $this->request->query])
````

**Comment by ADmad** [comment]
Also as I commented on related issue, usage like
```php
    ->find('search', $this->request->query)
```
will cause bad things to happen if query string contains other valid find options like `group` etc.

I don't want to open the possibility of users shooting themselves in the foot.

**Comment by josegonzalez** [comment]
@burzum friendly *bump* :)

**Comment by burzum** [comment]
@josegonzalez  well, @ADmad basically rejected the idea. So just close it or change it in another way.

**Comment by dereuromark** [comment]
We can do this now when we are fine with the 2.0.0 major bump.

**Comment by dereuromark** [comment]
So should be merged along with https://github.com/FriendsOfCake/search/pull/83

File: src/Model/Behavior/SearchBehavior.php [hunk 1]
```diff
@@ -45,8 +45,12 @@ class SearchBehavior extends Behavior
      */
     public function findSearch(Query $query, array $options)
     {
-        if (isset($options['search'])) {
-            $options = $options['search'];
+        if (!isset($options['_search']) && (!isset($options['_filter']) || $options['_filter'] === true)) {
+            $options = $this->filterParams($options);
```
Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: The check for `_search` and `_filter` should be mutually exclusive. `filterParams()` should be called after options are extracted from `$options['_search']` below.

Comment (burzum) [inline src/Model/Behavior/SearchBehavior.php]: Would you mind adding that change?

File: src/Model/Behavior/SearchBehavior.php [hunk 2]
```diff
@@ -45,8 +45,12 @@ class SearchBehavior extends Behavior
      */
     public function findSearch(Query $query, array $options)
     {
-        if (isset($options['search'])) {
```
Comment (jippi) [inline src/Model/Behavior/SearchBehavior.php]: why would we use a `_` prefix ?

**Comment by dereuromark** [comment]
So will this be included in the next major?

**Comment by ADmad** [comment]
Yes, I am making an alternate PR.

**Comment by burzum** [comment]
@ADmad why not just completing this one?

**Comment by ADmad** [comment]
Because starting fresh myself was more convenient for me.

**Pull request closed without merging**